contracts/contango/ContangoCauldron.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+
+import "@yield-protocol/utils-v2/contracts/math/WMul.sol";
+import "@yield-protocol/utils-v2/contracts/math/WDivUp.sol";
+import "../constants/Constants.sol";
+import "../Cauldron.sol";
+import "./util/EnumerableSet.sol";
+
+contract ContangoCauldron is Cauldron {
+    using CauldronMath for uint128;
+    using EnumerableSet for EnumerableSet.Bytes6Set;
+    using CastU256I256 for uint256;
+    using WMul for uint256;
+    using WDivUp for uint256;
+
+    EnumerableSet.Bytes6Set private assetsInUse;
+    mapping(bytes6 => DataTypes.Balances) public balancesPerAsset;
```

Comment:
Are we doing away with individual vaults completely, or having a protocol vault and individual vaults in parallel?

Reply:
We have both, this impl calls the `_pour` on the parent that does all the existent stuff, then it replaces the call to `_level` by its internal call to `_pour` to update the global values and do global collateralisation check.
Maybe we could rename the second `_pour` to something else?